app/views/LoginView/UserForm.tsx

```diff
@@ -0,0 +1,170 @@
+
+	const { colors } = useTheme();
+	const dispatch = useDispatch();
+
+	const getUser = () => getValues('user');
+	useImperativeHandle(ref, () => ({
+		getUser
+	}));
+
+	const {
+		Accounts_EmailOrUsernamePlaceholder,
+		Accounts_PasswordPlaceholder,
+		Accounts_PasswordReset,
+		Accounts_RegistrationForm_LinkReplacementText,
+		isFetching,
+		Accounts_RegistrationForm,
+		Site_Name,
+		inviteLinkToken
+	} = useAppSelector(state => ({
+		Accounts_RegistrationForm: state.settings.Accounts_RegistrationForm as string,
+		Accounts_RegistrationForm_LinkReplacementText: state.settings.Accounts_RegistrationForm_LinkReplacementText as string,
+		isFetching: state.login.isFetching,
+		Accounts_EmailOrUsernamePlaceholder: state.settings.Accounts_EmailOrUsernamePlaceholder as string,
+		Accounts_PasswordPlaceholder: state.settings.Accounts_PasswordPlaceholder as string,
+		Accounts_PasswordReset: state.settings.Accounts_PasswordReset as boolean,
+		Site_Name: state.settings.Site_Name as string,
+		inviteLinkToken: state.inviteLinks.token
+	}));
+
+	const showRegistrationButton =
+		Accounts_RegistrationForm === 'Public' || (Accounts_RegistrationForm === 'Secret URL' && inviteLinkToken?.length);
+
+	const register = () => {
+		navigation.navigate('RegisterView', { title: Site_Name });
+	};
+
+	const forgotPassword = () => {
+		navigation.navigate('ForgotPasswordView', { title: Site_Name });
+	};
+
+	const submit = ({ password, user }: ISubmit) => {
+		if (!isValid) {
+			return;
+		}
+		Keyboard.dismiss();
+		dispatch(loginRequest({ user, password }));
+	};
+
+	return (
+		<>
+			<Text style={[styles.title, sharedStyles.textBold, { color: colors.titleText }]}>{I18n.t('Login')}</Text>
+			<ControlledFormTextInput
+				name='user'
+				control={control}
+				label={I18n.t('Username_or_email')}
+				containerStyle={styles.inputContainer}
+				placeholder={Accounts_EmailOrUsernamePlaceholder || I18n.t('Username_or_email')}
+				keyboardType='email-address'
+				returnKeyType='next'
+				onSubmitEditing={() => {
+					passwordInput.current?.focus();
+				}}
+				testID='login-view-email'
+				textContentType='username'
+				autoComplete='username'
+			/>
+			<ControlledFormTextInput
+				name='password'
+				control={control}
+				label={I18n.t('Password')}
+				containerStyle={styles.inputContainer}
+				inputRef={e => {
+					passwordInput.current = e;
+				}}
+				placeholder={Accounts_PasswordPlaceholder || I18n.t('Password')}
+				returnKeyType='send'
+				secureTextEntry
+				onSubmitEditing={handleSubmit(submit)}
+				testID='login-view-password'
+				textContentType='password'
+				autoComplete='password'
+			/>
+			<Button
+				title={I18n.t('Login')}
+				type='primary'
+				onPress={handleSubmit(submit)}
+				testID='login-view-submit'
+				loading={isFetching}
+				disabled={!isValid}
+				style={styles.loginButton}
+			/>
+			{Accounts_PasswordReset && (
```

Comment:
```suggestion
			{Accounts_PasswordReset ? (
```
https://medium.com/geekculture/stop-using-for-conditional-rendering-in-react-a0f7b96200f8
https://koprowski.it/2020/conditional-rendering-react-native-text-crash/